### PR TITLE
docs: update retry plugin docs

### PR DESCRIPTION
**Pull request by ramaraochavali**
*Description*: Updated/Restructured the retry plugin docs so that it lists the built-in plugins explicitly instead of just in the examples.
*Risk Level*: Low
*Testing*: N/A
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

File: docs/root/intro/arch_overview/http_connection_management.rst [hunk 1]
```diff
@@ -48,39 +48,56 @@ table <arch_overview_http_routing>`. The route table can be specified in one of
...
+  * *envoy.retry_host_predicates.previous_hosts* : This will keep track of previously attempted hosts, and allows to choose hosts
+    that haven't been attempted already.
+  
+* :ref:`Priority Predicates<envoy_api_field_route.RouteAction.RetryPolicy.retry_priority>`: These predicates can
   be used to adjust the priority load used when selecting a priority for a retry attempt. Only one such
-  plugin may be specified.
+  predicate may be specified. Host selection will continue until either the configured predicate
+  accepts the host or a configurable
+  :ref:`max attempts <envoy_api_field_route.RouteAction.RetryPolicy.host_selection_retry_max_attempts>`
+  has been reached.
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply docs/root/intro/arch_overview/http_connection_management.rst]: My intention of repeating it is to highlight that both the predicates support that. I will move that to a separate paragraph

File: docs/root/intro/arch_overview/http_connection_management.rst [hunk 2]
```diff
@@ -48,39 +48,56 @@ table <arch_overview_http_routing>`. The route table can be specified in one of
...
+
+For example, to configure retries to prefer hosts that haven't been attempted already, the built-in
 ``envoy.retry_host_predicates.previous_hosts`` predicate can be used:
 
 .. code-block:: yaml
 
   retry_policy:
     retry_host_predicate:
-    - name: envoy.retry_host_predicates.previous_hosts
+      name: envoy.retry_host_predicates.previous_hosts
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply docs/root/intro/arch_overview/http_connection_management.rst]: Yeah. Sorry. Removed it by mistake.

**Comment by ramaraochavali** [comment]
@zuercher addressed the feedback. PTAL.

File: docs/root/intro/arch_overview/http_connection_management.rst [hunk 3]
```diff
@@ -48,24 +48,37 @@ table <arch_overview_http_routing>`. The route table can be specified in one of
...
+Normally during retries, hosts selection follows the same process as the original request. To modify 
+this behavior retry plugins can be used, which fall into two categories:
+
+* :ref:`Host Predicates <envoy_api_field_route.RouteAction.RetryPolicy.retry_host_predicate>`:
+  These predicates can be used to "reject" a host, which will cause host selection to be reattempted. 
+  Any number of these predicates can be specified, and the host will be rejected if any of the predicates reject the host. 
+
+  Envoy supports the following built-in host predicates
+
+  * *envoy.retry_host_predicates.previous_hosts* : This will keep track of previously attempted hosts, and rejects
```
Comment (zuercher) [inline docs/root/intro/arch_overview/http_connection_management.rst]: One last thing. Here and below remove the space before the colon.

File: docs/root/intro/arch_overview/http_connection_management.rst [hunk 4]
```diff
@@ -48,24 +48,37 @@ table <arch_overview_http_routing>`. The route table can be specified in one of
...
+    hosts that have already been attempted.
+  
+* :ref:`Priority Predicates<envoy_api_field_route.RouteAction.RetryPolicy.retry_priority>`: These predicates can
   be used to adjust the priority load used when selecting a priority for a retry attempt. Only one such
-  plugin may be specified.
+  predicate may be specified.
+
+  Envoy supports the following built-in priority predicates
+
+  * *envoy.retry_priority.previous_priorities* : This will keep track of previously attempted priorities,
```
Comment (zuercher) [inline docs/root/intro/arch_overview/http_connection_management.rst]: The second instance is here.

**Comment by ramaraochavali** [comment]
@zuercher Removed the spaces. should be good to go. PTAL.

**Pull request merged**